authservice/auth_service.go

```diff
@@ -1145,14 +1146,50 @@ type ServiceReg struct {
 	PubKey           *PubKey `json:"pub_key" bson:"pub_key"`
 }
 
+// -------------------- PrivKey --------------------
+
+// PrivKey represents a private key object including the key and related metadata
+type PrivKey struct {
+	Key    authutils.PrivateKey `json:"-" bson:"-"`
+	KeyPem string               `json:"key_pem" bson:"key_pem" validate:"required"`
+}
+
+// LoadKeyFromPem parses "KeyPem" and sets the "Key" and "Kid"
+func (p *PrivKey) LoadKeyFromPem() error {
```

Comment:
I think we decided that we wanted to have the caller tell us what type of key it is. In this case we can actually just include the type in the `PrivKey` struct above and use that when decoding the PEM. The same applies to the pubkey below.